news/12627-fix-msys2-conda.sh-paths

```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* <news item>
+
+### Bug fixes
+
+* Stop pre-converting paths to unix style on Windows in `conda.sh`. Instead rely on `cygpath` at runtime. (#12627)
```

Comment:
```suggestion
* Stop pre-converting paths to unix style on Windows in `conda.sh`, so that they are prefix replaceable upon installation, which got broken by #12509. It also relies on `cygpath` at runtime, which all msys2/cygwin bash versions on windows should have available. (#12627)
```